### ISSUE TITLE
Why is no-magic-number default "ignore" option set to [0, 1, 2]?

### ISSUE THREAD

**Issue by alberto** [eslint/eslint#4193]
It caught me by surprise that no-magic-number ignores numbers `[0, 1, 2]` by default. That's pretty magic if you ask me :). 

If there is a rationale behind that (I can't think of any), it should be documented in the rule. Otherwise it should not ignore any number by default.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by hzoo** [comment]
Yeah maybe it shouldn't have a default and just require an array

**Comment by nzakas** [comment]
It is documented: http://eslint.org/docs/rules/no-magic-numbers#ignore

The rationale is that these are the numbers people use most frequently, such as:

```js
if (items.length === 0) {}
if (items.length > = 1) {}
```

Etc.

Honestly, I didn't catch this on code review so I was a bit surprised too, but I think it makes sense.

If we want to make a change, it will have to wait for 2.0.0.

**Comment by alberto** [comment]
I meant the reason should be documented. 
IMO `0` could get a pass, `1` is dubious, but `2`?

**Comment by nzakas** [comment]
Fair point. @kombucha what's the rationale for each of the default allowed magic numbers?

**Comment by kombucha** [comment]
I mimicked the default behaviour of buddy.js.
I see know that it was just `[0, 1]`, `[0, 1, 2]` was just an example in the docs.
And when I look at the code of buddy.js, the defaults seems to actually be an empty array...

**Comment by nzakas** [comment]
Okay, let's plan on changing this for 2.0.0.

**Comment by kombucha** [comment]
I guess an empty array as a default might make more sense, and just let people decide what make sense in their project.
Or 0 and -1, since they're so pervasive when manipulating arrays.

**Comment by nzakas** [comment]
I'd go with an empty array as the default - let people add what they want and then there's no controversy.

**Comment by alberto** [comment]
@nzakas +1 on that

Can this be tackled now, or do you prefer to wait?

**Comment by nzakas** [comment]
Best to wait until we're ready for 2.0.0. I hate having PRs sitting around.
